gpu/gpu.go

```diff
@@ -634,3 +628,32 @@ func (l GpuInfoList) GetVisibleDevicesEnv() (string, string) {
 		return "", ""
 	}
 }
+
+func GetDepDir() string {
+	// On Windows/linux we bundle the dependencies at the same level as the executable
+	appExe, err := os.Executable()
+	if err != nil {
+		slog.Warn("failed to lookup executable path", "error", err)
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		slog.Warn("failed to lookup working directory", "error", err)
+	}
+	// Scan for any of our dependeices, and pick first match
+	for _, root := range []string{filepath.Dir(appExe), cwd} {
```

Comment:
I don't have a strong opinion here, but would it be simpler if we didn't nest libraries within `cuda`, `rocm`, `oneapi` etc vs just keeping a flat directory with all `.so` or `.dll` files Ollama may need? Assuming no conflicts (and maybe ROCm has a bunch of other files)

Reply:
Yes, I think this could work.

Reply:
done.